sdk/webpubsub/azure-messaging-webpubsubclient/azure/messaging/webpubsubclient/_client.py

```diff
@@ -558,40 +693,50 @@ def _retry(self, func: Callable[[], None]):
                 )
                 time.sleep(delay_seconds)
 
-    def _call_back(self, callback_type: Union[CallbackType, str], *args):
+    def _call_back(self, callback_type: CallbackType, *args):
         for func in self._handler[callback_type]:
             # _call_back works in listener thread which must not be blocked so we have to execute the func
             # in new thread to avoid dead lock
             threading.Thread(target=func, args=args, daemon=True).start()
 
-    def _start_from_restarting(self):
+    def _start_from_restarting(self, reconnect_tried_times: Optional[int] = None):
         if self._state != WebPubSubClientState.DISCONNECTED:
             _LOGGER.warning("Client can be only restarted when it's Disconnected")
             return
 
         try:
-            self._start_core()
+            self._start_core(reconnect_tried_times)
         except Exception as e:
             self._state = WebPubSubClientState.DISCONNECTED
             raise e
 
-    def _handle_auto_reconnect(self):
+    def _handle_auto_reconnect(self, reconnect_tried_times: Optional[int] = None):
         _LOGGER.debug("start auto reconnect")
         success = False
-        attempt = 0
+        attempt = -1 if reconnect_tried_times is None else reconnect_tried_times
         while not self._is_stopping:
             try:
-                self._start_from_restarting()
+                attempt = attempt + 1
```

Comment:
Is the only reason we count attempts for calculating the delay?

Reply:
Yep.